yaml-tests/src/test/resources/table-functions.yamsql

```diff
@@ -0,0 +1,131 @@
+      - result: [{B: 1, C: 2.0, W: [42, 43, 44]},
+                 {B: 11, C: 3.0, W: [420, 430, 440]}]
+    -
+      - query: select * from values (1, 2.0, [('a', 'b', [1, 2, 3])]), (11, 3.0, [('d', 'e', [10, 20, 30])])  as A(B, C, W(X, Y, Z))
+      - result: [{B: 1, C: 2.0, W: [{X: 'a', Y: 'b', Z: [1, 2, 3]}]},
+                 {B: 11, C: 3.0, W: [{X: 'd', Y: 'e', Z: [10, 20, 30]}]}]
+    -
+      - query: select A.B, C, W from values (1, 2.0, (3, 4, 'foo')), (10, 90.2, (5, 6.0, 'bar')) as A(B, C, W(X, Y, Z))
+      - explain: "EXPLODE array((@c12 AS B, @c14 AS C, (@c17 AS X, promote(@c19 AS DOUBLE) AS Y, @c21 AS Z) AS W), (@c26 AS B, @c28 AS C, (@c31 AS X, @c33 AS Y, @c35 AS Z) AS W)) | MAP (_.B AS B, _.C AS C, _.W AS W)"
+      - result: [{B: 1, C: 2.0, W: {X: 3, Y: 4.0, Z: 'foo'}},
+                 {B: 10, C: 90.2, W: {X: 5, Y: 6.0, Z: 'bar'}}]
+    -
+      - query: select A.B, C as Q, W.X from values (1, 2.0, (3, 4, 'foo')), (10, 90.2, (5, 6.0, 'bar')) as A(B, C, W(X, Y, Z))
+      - explain: "EXPLODE array((@c16 AS B, @c18 AS C, (@c21 AS X, promote(@c23 AS DOUBLE) AS Y, @c25 AS Z) AS W), (@c30 AS B, @c32 AS C, (@c35 AS X, @c37 AS Y, @c39 AS Z) AS W)) | MAP (_.B AS B, _.C AS Q, _.W.X AS X)"
+      - result: [{B: 1, Q: 2.0, X: 3},
+                 {B: 10, Q: 90.2, X: 5}]
+    -
+      - query: select * from (select A.B, C as Q, W.X from values (1, 2.0, (3, 4, 'foo')), (10, 90.2, (5, 6.0, 'bar')) as A(B, C, W(X, Y, Z))) as u
+      - explain: "EXPLODE array((@c20 AS B, @c22 AS C, (@c25 AS X, promote(@c27 AS DOUBLE) AS Y, @c29 AS Z) AS W), (@c34 AS B, @c36 AS C, (@c39 AS X, @c41 AS Y, @c43 AS Z) AS W)) | MAP (_.B AS B, _.C AS Q, _.W.X AS X)"
+      - result: [{B: 1, Q: 2.0, X: 3},
+                 {B: 10, Q: 90.2, X: 5}]
+    -
+      - query: select * from (select A.B, C as Q, W.X from values (1, 2.0, (3, 4, 'foo')), (10, 90.2, (5, 6.0, 'bar')) as A(B, C, W(X, Y, Z))) as u where b < 8
+      - explain: "EXPLODE array((@c20 AS B, @c22 AS C, (@c25 AS X, promote(@c27 AS DOUBLE) AS Y, @c29 AS Z) AS W), (@c34 AS B, @c36 AS C, (@c39 AS X, @c41 AS Y, @c43 AS Z) AS W)) | MAP (_.B AS B, _.C AS Q, _.W.X AS X) | FILTER _.B LESS_THAN @c68"
+      - result: [{B: 1, Q: 2.0, X: 3}]
+    -
+      - query: select * from range(1, 4)
+      - result: [{ID: 1}, {ID: 2}, {ID: 3}]
+    -
+      - query: select * from range(-1)
+      - error: XXXXX
+    -
+      - query: select * from range(-1, 4)
+      - error: XXXXX
+    -
+      - query: select * from range(1, 4, -1)
+      - error: XXXXX
+    -
+      - query: select * from range(1, 4, 0)
+      - error: XXXXX
+    -
+      - query: select * from range(0, 12, 5)
+      - result: [{ID: 0}, {ID: 5}, {ID: 10}]
+    -
+      - query: select * from range(0, 11, 5)
+      - explain: "TFunc range()"
+      - result: [{ID: 0}, {ID: 5}, {ID: 10}]
+    -
+      - query: select * from range(6 - 6, 14 + 6 + 1, 20 - 10)
+      - explain: "TFunc range()"
```

Comment:
`range()` needs to explain its arguments

Reply:
I had a todo there, but fixed it in the PR.